non-obvious regular expressions

By default library consider expectedJson as a regular expression. But sometimes its not obvious that brackets in json string values may be recognized as a regular expression characters. Lets say, we have the following json:
`
    [{ "name" : "someText (anotherText)", "code" : "oneMoreText" }]
`
If we try to compare this json with itself, 
`
JSONCompare.assertEquals(null, json, json, null, new CompareMode[0]);
`
we get an exception message:
`
Exception in thread "main" java.lang.AssertionError: Expected element from position 1 was NOT FOUND:
{ "name" : "someText (anotherText)", "code" : "oneMoreText" }
Expected:
[{ "name" : "someText (anotherText)", "code" : "oneMoreText" }]
But got:
[{ "name" : "someText (anotherText)", "code" : "oneMoreText" }]
` 
which is absolutely confusing.

Maybe it would be convenient, if exception message mentions that comparison failed due to regular expression misuse in such cases when actualJson contains the same 'regular expression' as in expectedJson

--- a/Changelog.md
+++ b/Changelog.md
@@ -1,7 +1,10 @@
 # Changelog
 
-## 2.40-SNAPSHOT
+## 2.41-SNAPSHOT
 
+
+## 2.40
+Enhance AssertionError message with hint for unintentional regexes  
 
 ## 2.39
 Update dependencies  

--- a/src/main/java/ro/skyah/comparator/JSONCompare.java
+++ b/src/main/java/ro/skyah/comparator/JSONCompare.java
@@ -23,6 +23,9 @@ public class JSONCompare {
 
     private static final ObjectMapper MAPPER = new ObjectMapper().enable(DeserializationFeature.FAIL_ON_TRAILING_TOKENS);
 
+    private static final String ASSERTION_ERROR_HINT_MESSAGE = "Hint: By default, json matching uses regular expressions.\n" +
+            "If expected json contains unintentional regexes, then quote them between \\Q and \\E delimiters or use a custom comparator.";
+
     public static void assertEquals(String expected, String actual, CompareMode... compareModes) {
         assertEquals(null, expected, actual, compareModes);
     }
@@ -105,8 +108,8 @@ public class JSONCompare {
                     comparator == null ? new DefaultJsonComparator() : comparator,
                     new HashSet<>(Arrays.asList(compareModes))).match();
         } catch (MatcherException e) {
-            String defaultMessage = String.format("%s\nExpected:\n%s\nBut got:\n%s ", e.getMessage(),
-                    prettyPrint(expected), MessageUtil.cropL(prettyPrint(actual)));
+            String defaultMessage = String.format("%s\nExpected:\n%s\nBut got:\n%s\n\n%s", e.getMessage(),
+                    prettyPrint(expected), MessageUtil.cropL(prettyPrint(actual)), ASSERTION_ERROR_HINT_MESSAGE);
             fail(message == null ? defaultMessage : defaultMessage + "\n" + message);
         }
     }
@@ -123,6 +126,14 @@ public class JSONCompare {
         fail(message == null ? defaultMessage : defaultMessage + "\n" + message);
     }
 
+    public static String prettyPrint(JsonNode jsonNode) {
+        try {
+            return MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(jsonNode);
+        } catch (JsonProcessingException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
     private static JsonNode getJson(String json) {
         JsonNode jsonNode = null;
         try {
@@ -131,13 +142,5 @@ public class JSONCompare {
             fail(String.format("Not a JSON:\n%s", MessageUtil.cropL(json)));
         }
         return jsonNode;
-    }
-
-    public static String prettyPrint(JsonNode jsonNode) {
-        try {
-            return MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(jsonNode);
-        } catch (JsonProcessingException e) {
-            throw new RuntimeException(e);
-        }
     }
 }

--- a/src/test/java/ro/skyah/comparator/issues/Issue13Test.java
+++ b/src/test/java/ro/skyah/comparator/issues/Issue13Test.java
@@ -4,6 +4,7 @@ import org.junit.jupiter.api.Test;
 import ro.skyah.comparator.JSONCompare;
 
 public class Issue13Test {
+    //Issue: https://github.com/fslev/json-compare/issues/28
 
     @Test
     public void testComparisonOfJsonWithUnintendedRegex() {

--- a/src/test/java/ro/skyah/comparator/issues/Issue13Test.java
+++ b/src/test/java/ro/skyah/comparator/issues/Issue13Test.java
@@ -1,0 +1,14 @@
+package ro.skyah.comparator.issues;
+
+import org.junit.jupiter.api.Test;
+import ro.skyah.comparator.JSONCompare;
+
+public class Issue13Test {
+
+    @Test
+    public void testComparisonOfJsonWithUnintendedRegex() {
+        String expected = "[{ \"name\" : \"someText (anotherText)\", \"code\" : \"oneMoreText\" }]";
+        String actual = "[{ \"name\" : \"someText (anotherText)\", \"code\" : \"oneMoreText\" }]";
+        JSONCompare.assertEquals(expected, actual);
+    }
+}

--- a/src/test/java/ro/skyah/comparator/issues/Issue13Test.java
+++ b/src/test/java/ro/skyah/comparator/issues/Issue13Test.java
@@ -3,13 +3,20 @@ package ro.skyah.comparator.issues;
 import org.junit.jupiter.api.Test;
 import ro.skyah.comparator.JSONCompare;
 
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
 public class Issue13Test {
     //Issue: https://github.com/fslev/json-compare/issues/28
 
     @Test
-    public void testComparisonOfJsonWithUnintendedRegex() {
+    // Check for hint message from failed comparison of jsons with unintentional regexes
+    public void compareJsonWithUnintentionalRegex() {
         String expected = "[{ \"name\" : \"someText (anotherText)\", \"code\" : \"oneMoreText\" }]";
         String actual = "[{ \"name\" : \"someText (anotherText)\", \"code\" : \"oneMoreText\" }]";
-        JSONCompare.assertEquals(expected, actual);
+        try {
+            JSONCompare.assertEquals(expected, actual);
+        } catch (AssertionError e) {
+            assertTrue(e.getMessage().contains("Hint"));
+        }
     }
 }

